assets/javascripts/modules/checkout/fieldSwitcher.js

```diff
@@ -177,6 +175,10 @@ define([
         };
 
         var refresh = function () {
+            if (determinesPriceBanding) {
+                $('.js-currency-override-checkbox').attr('checked', false);
+            }
+            
             var currentState = getCurrentState();
```

Comment:
we need to reset the checkbox state before getting the current state otherwise it will always return the currency as GBP if it is checked